### ISSUE TITLE
Vanir CM-13 ROM patching failed

### ISSUE THREAD

**Issue by pejakm** [nagalun/needle-zip#2]
The rom is here: http://www.modaco.com/forums/topic/377558-vanir-marshmallow-featuring-mourta-kernel-hybrid-2016-07-27/

Flashing from recovery (TWRP 3.0.2) failed. I've tried manually from recovery (using console), the only error is this:

> **\* Disassembling framework.jar...
> /tmp/updater: line 29: can't create /proc/self/fd/none:
> nonexistant directory
> !!! Something weird happened, exiting...

I'll try to patch directly from within running system and report back.

**Comment by nagalun** [comment]
I need to re-do the installation script as it is buggy and doesn't do many error work-arounds, I haven't got around to do that because I would like to get it working with 64 bit devices too, which I don't have any.

Did it tell you anything different when flashing (normally)?
The problem is this bit right here:

```
> /proc/self/fd/$FD
```

I'll fix it later
For now, try this zip when instaling in a running system:
https://github.com/nagalun/needle-zip/blob/507ab6503df4d296db63bf12c3fade70965c7a0f/needle-fakesig.zip

Before running the script do these commands:
mount -o rw,remount /
mount -o rw,remount /system

**Comment by pejakm** [comment]
I am sorry, I had to revert my backup of a working system. I did try manually, I unpacked zip file, run update-binary, but this time it complained about not able to write into /tmp. System was mounted rw, zip file and binary was in /cache. After running the command (`sh /cache/update-binary none none /cache/needle-fakesig.zip`) zip file was unpacked in /cache, but the script complained it cannot write into the /tmp and it cannot find anything under /tmp/needle(-fakesig).

**Comment by nagalun** [comment]
Yeah because you need to make both /system and / (rootfs) read write, try mkdir /tmp too, after that

**Comment by pejakm** [comment]
> Yeah because you need to make both /system and / (rootfs) read write, try mkdir /tmp too, after that

Oh damn, didn't think of that. :smile_cat: :sweat_smile: Thanks, I'm gonna try today or tomorrow again, and will report back.

**Comment by pejakm** [comment]
Nagalun, I'm sorry, I didn't try the Vanir rom again. In the meantime, I got myself a new phone on which your script works (although reported an error), thus making this bug unimportant to me. Feel free to close it if you want, I wont be able to provide info anymore.

**Comment by nagalun** [comment]
No, it's fine, this has many problems that I'm lazy to fix because it works on my phone. Here is a tip, if you flash the zip after first boot (dalvik/art-cache is built) it will take much less time than if you do it right after flashing an update/wiping cache
